core/src/cluster_info_vote_listener.rs

```diff
@@ -389,17 +409,14 @@ impl ClusterInfoVoteListener {
             let would_be_leader = poh_recorder
                 .lock()
                 .unwrap()
-                .would_be_leader(20 * DEFAULT_TICKS_PER_SLOT);
+                .would_be_leader(3 * slot_hashes::MAX_ENTRIES as u64 * DEFAULT_TICKS_PER_SLOT);
             if let Err(e) = verified_vote_packets.receive_and_process_vote_packets(
                 &verified_vote_label_packets_receiver,
-                &mut update_version,
                 would_be_leader,
             ) {
                 match e {
-                    Error::CrossbeamRecvTimeout(RecvTimeoutError::Disconnected) => {
-                        return Ok(());
-                    }
-                    Error::CrossbeamRecvTimeout(RecvTimeoutError::Timeout) => (),
+                    Error::CrossbeamRecvTimeout(RecvTimeoutError::Disconnected)
+                    | Error::ReadyTimeout => (),
```

Comment:
previously if channel is disconnected, _this_ thread will exist with `Ok`, is it intended to keep _this_ thread running even _receiver_ has gone?

Reply:
ah this was a bug, good catch!